cli/args/flags.rs

```diff
@@ -317,7 +317,7 @@ pub struct Flags {
   pub lock: Option<PathBuf>,
   pub log_level: Option<Level>,
   pub no_remote: bool,
-  pub no_lock: bool,
+  pub no_lock: Option<bool>,
```

Comment:
It seems like there's no need to make this optional since it seems there are only two states that need to be represented and not three?

Reply:
We cannot know when to override the `lock` of the config file if we don’t use an option.

Reply:
If there's only two states then this can be represented with a `bool`, but if there's three then this should indeed be an option. It doesn't seem possible to do `--no-lock=false` and it looks like the only states are "flag present" and "flag not present", so I believe we only need to use a bool? Then we can override the `lock` of the config file when it's `true`.

Reply:
You can, by first checking if it was configured via a config file. I agree with David that it should be left as a `bool`

Reply:
Done!